tomography_preprocessing/denoising/care4relion/care4relion.py

```diff
@@ -0,0 +1,149 @@
+import starfile
+import rich
+import typer
+import subprocess
+from rich.progress import track
+
+from .train_neural_network import train_neural_network
+from ._utils import *
+from .constants import predict_config_prefix, even_suffix 
+from .._cli import cli
+from ...utils.relion import relion_pipeline_job
+
+console = rich.console.Console(record=True)
+
+@cli.command(name='Care4Relion')
+@relion_pipeline_job
+def care4relion(
+    tilt_series_star_file: Path = typer.Option(...),
+    output_directory: Path = typer.Option(...),
+    training_tomograms: Optional[str] = typer.Option(None),
+    model_name: Optional[Path] = typer.Option(None),
+    n_tiles: Optional[Tuple[int,int,int]] = typer.Option((1,1,1))
+
+):
+    """Denoises tomograms using cryoCARE (Thorsten Wagner version) (https://github.com/thorstenwagner/cryoCARE_mpido)
+    
+    Requires that two tomograms have been generated using the same sample. These can be generated via taking odd/even 
+    frames during Motion Correction (optimal) or by taking odd/even tilts during tomogram reconstruction.
+    The location of these tomograms should be specified in the global star file for all tilt series with the headers: 
+    
+     
+    _rlnTomoTomogramHalvesForDenoisingEven
+    
+      _rlnTomoTomogramHalvesForDenoisingOdd
+
+    Parameters
+    ----------
+    
+    tilt_series_star_file: RELION tilt-series STAR file.
+    
+    output_directory: directory in which results will be stored.
+    
+    training_tomograms: tomograms which are to be used for denoising neural network training.
+        User should enter tomogram names as rlnTomoName, separated by colons, e.g. TS_1:TS_2
+	
+    model_name (optional): if you have already trained a denoising neural network and want to skip the training step,
+        user can provide the path to the model.tar.gz produced by a previous cryoCARE denoising job. This will skip
+        straight to tomogram generation.
+	
+    n-tiles (optional): Initial tiles per dimension during prediction step. Should get increased \ 
+        if the tiles do not fit on the GPU. However, sometimes this parameter is a source of errors causing out of memory \
+        problems and tomogram dimensions to be wrong. We have found other useful values for this include 4,4,2 and 2,4,4 \
+        (default = 1,1,1). Enter as: `--n-tiles 4 4 2`
+        
+
+    Returns
+    -------
+    Denoised tomograms.
+    """
+    if not tilt_series_star_file.exists():
+        e = 'Could not find tilt series star file'
+        console.log(f'ERROR: {e}')
+        raise RuntimeError(e)    
+    
+    if model_name is None and training_tomograms is None:
+        e = 'Both model name and training tomograms are empty. Either provide a trained network through model name or specify \
+            which tomograms to train on (using rlnTomoName of the tilt series, separated by colons) in training tomograms'
+        console.log(f'ERROR: {e}')
+        raise RuntimeError(e)
```

Comment:
see above comment about 'doing too much'

Reply:
as above, will separate training and prediction